### DISCUSSION TITLE
Intellisense doesn't work

### DISCUSSION THREAD

**Discussion by xLipeQ** [AvaloniaUI/Avalonia#11423] [Q&A]
I ran to a big problem that I don't have any preview or intellisense in Visual Studio 2022 axaml file. Every avalonia project don't have that. I can change the code and run it. When I installed avalonia at first it used to work fine, but suddenly it stopped. 

I've seen this one, but i don't know where to find samples\directory.build.props
https://github.com/AvaloniaUI/Avalonia/issues/3966

**Comment by BinToss** [comment]
> I ran to a big problem that I don't have any preview or intellisense in Visual Studio 2022 axaml file. Every avalonia project don't have that. I can change the code and run it. When I installed avalonia at first it used to work fine, but suddenly it stopped. 
> 
> I've seen this one, but i don't know where to find samples\directory.build.props
> https://github.com/AvaloniaUI/Avalonia/issues/3966

Which version of Avalonia are you using?

**Comment by zBugi** [comment]
Hi, I ran into the same issue.
I am using Avalonia 11.0.6 and Visualstudio 2022. 
After the Update of the "Avalonia for Visual Studio 2022" extension, the extension was Disabled.
I enabled it then restarted Visual Studio, then I right clicked on an opened .xaml file and chosed View Design.
After that it worked again.

**Comment by kyuranger** [comment]
I met the same problem after upgrading to 11.0.6.

**Reply by BinToss** [reply]
As noted in this project's README.md, this extension only works with Avalonia v11.0.0 through v11.0.4. Avalonia v11.0.5 introduced a change that broke away from what this extension expects.

Ideally, extensions like this one should be gathering information from a project's Avalonia version and the restored Avalonia package from the local NuGet cache.
This may be implemented by getting workspace information from the official C# extension or by inspecting a built project's project.deps.json (dependency tree) and inspecting the Avalonia version(s?) and dependents. Then, search the NuGet cache for those packages and do one or all of the following: download source code via SourceLink metadata, download NuGet symbols package if available, or decompile to source code.
The last option can be accomplished by two VSCode extensions and/or a NuGet package.
- `ms-dotnettools.csdevkit` (decomp feature disabled by default)
- `icsharpcode.ilspy-vscode` (reads from MSIL assembly file)
- `ICSharpCode.Decompiler`

**Reply by vlachdev** [reply]
Where did you find information about supported version of Avalonia?

So i can check before updating in future, currently downgrading to `v11.0.4` helped, but i have no idea what version is currently supported.

Thanks

**Reply by BinToss** [reply]
If you depend on Avalonia 11.0.0 - 11.0.4, use AvaloniaVS 11.0.4.
If you depend on Avalonia 11.0.5 or later, use the latest AvaloniaVS 11.0.5 or later.


I...don't remember where I got this information. I know some of it came from the corresponding issue in AvaloniaVSCode where it was stated by @prashantvc. The changes to CompletionEngine between AvanloniaVS 11.0.4 and 11.0.5 don't make it obvious why IntelliSense broke. Maybe something in the XmlParser needed to be changed due to an accidental breaking change in AvaloniaUI?

Edit: somewhere under IntermediateOutput (a project's `obj` folder), there's a `references` plain text file with all of the project's assembly references. These are used by the CompletionEngine (used by AvaloniaVS, AvaloniaVSCode, and AvaloniaRider) to load assemblies and provide completion results.

In .NET 8, the IntermediateOutput path was no longer hard coded to a relative path (see [Artifacts Output Layout](https://learn.microsoft.com/en-us/dotnet/core/sdk/artifacts-output)). This isn't the only cause of the breaking change.

Avalonia 11.0.5 released with changes necessary to account for this. The CompletionEngine then also needed to account for this particular change in AvaloniaUI.

**Comment by maxkatz6** [comment]
Please create issues on [AvaloniaVS](https://github.com/AvaloniaUI/AvaloniaVS) repository instead.

**Comment by dev-ruben-lopez** [comment]
Same problem, in my case, VS2022,  SOLVED: what I did was to disable extensions for Xamarin and WPF that I had previously installed.

**Comment by malex81** [comment]
I have some problem - IntelliSence shows some nonsense. Extension version of Avalonia is 11.11.1.3. As IDE i use MS Visual Studio 2022
